File: problem_solve/07-22/[PGS] 폰켓몬/ye0ngeun.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        
        Set<Integer> types = new HashSet<>();
        
        for (int num : nums) {
            types.add(num);
        }
        
        return Math.min(nums.length / 2, types.size());
    }
}

File: problem_solve/09-03/[PGS] 점프와 순간 이동/ye0ngeun.py
def solution(n):
    answer = 0

    while n > 0:
        # 짝수면 순간이동으로 가능
        if n % 2 == 0:
            n //= 2
        # 홀수면 점프해야됨
        else:
            n -= 1
            answer += 1

    return answer
